vcsclient/fs.go

```diff
@@ -200,6 +197,28 @@ func GetFileWithOptions(fs vfs.FileSystem, path string, opt GetFileOptions) (*Fi
 	return &fwr, nil
 }
 
+// readDir uses the passed vfs.FileSystem to read from starting at the base path. If
+// shouldRecurse is set to true, it will recurse into all sub-folders and return the
+// full sub-tree.
+func readDir(fs vfs.FileSystem, base string, shouldRecurse bool) ([]*TreeEntry, error) {
```

Comment:
Did you consider using [`fs.WalkFS`](https://sourcegraph.com/github.com/kr/fs@master/.GoPackage/github.com/kr/fs/.def/WalkFS) for recursive FS walking? There might be some edge cases that that handles; I'm not sure.